modules/programs/lieer.nix

```diff
@@ -23,30 +23,231 @@ let
+        Remove local messages that have been deleted on the remote.
+      '';
+    };
+
+    replace_slash_with_dot = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Replace '/' with '.' in Gmail labels.
+      '';
+    };
+
+    timeout = mkOption {
+      type = types.ints.unsigned;
+      default = 600;
+      description = ''
+        HTTP timeout in seconds. 0 means forever or system timeout.
+      '';
     };
   };
 
+  syncOpts = {
+    enable = mkEnableOption "lieer synchronization service";
+
+    frequency = mkOption {
+      type = types.str;
+      default = "*:0/5";
+      description = ''
+        How often to synchronize the account.
+        </para><para>
+        This value is passed to the systemd timer configuration as the
+        onCalendar option. See
+        <citerefentry>
+          <refentrytitle>systemd.time</refentrytitle>
+          <manvolnum>7</manvolnum>
+        </citerefentry>
+        for more information about the format.
+      '';
+    };
+  };
+
+  lieerOpts = {
+    enable = mkEnableOption "lieer Gmail synchronization for notmuch";
+
+    notmuchSetupWarning = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        Warn if Notmuch is not also enabled for this account.
+        </para><para>
+        This can safely be disabled if <command>notmuch init</command>
+        has been used to configure this account outside of Home
+        Manager.
+      '';
+    };
+
+    settings = mkOption {
+      type = types.submodule {
+        freeformType = settingsFormat.type;
+        options = settingsOpts;
+      };
+      default = { };
+      description = ''
+        Settings which are applied to <filename>.gmailieer.json</filename>
+        for the account.
+        </para><para>
+        See the <link xlink:href="https://github.com/gauteh/lieer/">lieer manual</link>
+        for documentation of settings not explicitly covered by this module.
+      '';
+    };
+
+    sync = syncOpts;
+  };
+
+  lieerModule = types.submodule {
+    imports = [
+      (mkRenamedOptionModule [ "lieer" "dropNonExistingLabels" ] [
+        "lieer"
+        "settings"
+        "drop_non_existing_label"
+      ])
+      (mkRenamedOptionModule [ "lieer" "ignoreTagsRemote" ] [
+        "lieer"
+        "settings"
+        "ignore_remote_labels"
+      ])
+      (mkRenamedOptionModule [ "lieer" "ignoreTagsLocal" ] [
+        "lieer"
+        "settings"
+        "ignore_tags"
+      ])
+      (mkRenamedOptionModule [ "lieer" "timeout" ] [
+        "lieer"
+        "settings"
+        "timeout"
+      ])
+      (mkRenamedOptionModule [ "lieer" "replaceSlashWithDot" ] [
+        "lieer"
+        "settings"
+        "replace_slash_with_dot"
+      ])
+    ];
+
+    options = {
+      lieer = lieerOpts;
+
+      warnings = mkOption {
+        type = types.listOf types.str;
+        default = [ ];
+        internal = true;
+        visible = false;
+      };
```

Comment:
This is strange, but using `mkRenamedOptionModule` generates a `warnings` value on this submodule's config, and `warnings` is not a built-in config value for submodules.